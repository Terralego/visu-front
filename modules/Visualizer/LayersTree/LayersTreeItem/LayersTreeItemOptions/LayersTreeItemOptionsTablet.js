import React from 'react';
import classnames from 'classnames';
import { Button } from '@blueprintjs/core';

import translateMock from '../../../../../utils/translate';

import LayerFetchValues from '../LayerFetchValues';
import FiltersPanel from '../FiltersPanel';

const LayersTreeItemOptionsTablet = ({
  hasSomeOptionActive,
  isOptionsOpen,
  handleOptionPanel,
  layer,
  toggleFilters,
  isFilterVisible,
  getFilterPanelRef,
  form,
  toggleTable,
  isTableActive,
  displayTableButton,
  toggleWidgets,
  widgets,
  isWidgetActive,
  translate = translateMock({
    'layerstree.itemOptions.widget.open': 'open {{widget}}',
    'layerstree.itemOptions.widget.close': 'close {{widget}}',
    'layerstree.itemOptions.table.label': 'table',
    'layerstree.itemOptions.table.open': 'open table',
    'layerstree.itemOptions.table.close': 'close table',
    'layerstree.itemOptions.filter.label': 'filters',
    'layerstree.itemOptions.filter.open': 'open filters',
    'layerstree.itemOptions.filter.close': 'close filters',
    'layerstree.itemOptions.options.label': 'options',
    'layerstree.itemOptions.options.open': 'open options',
    'layerstree.itemOptions.options.close': 'close options',
  }),
}) => (
  <div
    className={classnames(
      'layerstree-node-content__options',
      'layerstree-node-content__options--mobile',
      { 'layerstree-node-content__options--active': hasSomeOptionActive },
    )}
  >
    {(widgets && !!widgets.length) && (
      widgets.map(widget => (
        <Button
          key={widget.component}
          className={classnames({
            'layerstree-node-content__options__button': true,
            'layerstree-node-content__options__button--active': isWidgetActive(widget),
          })}
          onClick={toggleWidgets(widget)}
          minimal
          icon="selection"
          title={`widget ${widget.component}`}
          alt={translate(`layerstree.itemOptions.widget.${isWidgetActive(widget) ? 'close' : 'open'}`, { widget: widget.component })}
        >
          {widget.component}
        </Button>
      ))
    )
}
    {displayTableButton && (
    <Button
      className={classnames(
        'layerstree-node-content__options__button',
        { 'layerstree-node-content__options__button--active': isTableActive },
      )}
      onClick={toggleTable}
      minimal
      icon="th"
      title={translate('layerstree.itemOptions.table.title')}
      alt={translate(`layerstree.itemOptions.table.${isTableActive ? 'close' : 'open'}`)}
    >
      {translate('layerstree.itemOptions.table.label')}
    </Button>
    )}
    {form && (
      <FiltersPanel
        visible={isFilterVisible}
        onMount={getFilterPanelRef}
        layer={layer}
      >
        {isFilterVisible && (
          <LayerFetchValues layer={layer} isFilterVisible={isFilterVisible} />
        )}
        <Button
          className={
          classnames(
            'layerstree-node-content__options__button',
            { 'layerstree-node-content__options__button--active': isFilterVisible },
          )}
          onClick={toggleFilters}
          minimal
          icon="filter"
          title="filter"
        >
        filtres
        </Button>
      </FiltersPanel>
    )}
    <Button
      className={classnames(
        'layerstree-node-content__options__button',
        'layerstree-node-content__options__button--more',
        { 'layerstree-node-content__options__button--active': isOptionsOpen },
      )}
      icon="more"
      minimal
      onClick={handleOptionPanel}
      title={translate('layerstree.itemOptions.options.label')}
      alt={translate(`layerstree.itemOptions.options.${isOptionsOpen ? 'close' : 'open'}`)}
    >
      {translate('layerstree.itemOptions.options.label')}
    </Button>
  </div>
);
export default LayersTreeItemOptionsTablet;
