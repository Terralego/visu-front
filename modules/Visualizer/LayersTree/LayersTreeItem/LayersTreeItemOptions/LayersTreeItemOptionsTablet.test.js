import React from 'react';
import renderer from 'react-test-renderer';
import LayersTreeItemOptionsTablet from '.';

it('should render correctly', () => {
  const tree = renderer.create((
    <LayersTreeItemOptionsTablet />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
