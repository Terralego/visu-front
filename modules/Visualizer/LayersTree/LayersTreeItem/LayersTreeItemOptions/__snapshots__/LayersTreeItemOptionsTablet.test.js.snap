// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
Array [
  <div
    className="layerstree-node-content__options layerstree-node-content__options--mobile"
  >
    <button
      className="bp3-button bp3-minimal layerstree-node-content__options__button layerstree-node-content__options__button--more"
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      title="options d'affichage"
      type="button"
    >
      <span
        className="bp3-icon bp3-icon-more"
        icon="more"
      >
        <svg
          data-icon="more"
          height={16}
          viewBox="0 0 16 16"
          width={16}
        >
          <desc>
            more
          </desc>
          <path
            d="M2 6.03a2 2 0 1 0 0 4 2 2 0 1 0 0-4zM14 6.03a2 2 0 1 0 0 4 2 2 0 1 0 0-4zM8 6.03a2 2 0 1 0 0 4 2 2 0 1 0 0-4z"
            fillRule="evenodd"
          />
        </svg>
      </span>
      <span
        className="bp3-button-text"
      >
        options
      </span>
    </button>
  </div>,
  <div
    className="layerstree-node-content__options layerstree-node-content__options--mobile"
  >
    <p>
      LayerFetchValues
    </p>
    <button
      className="bp3-button bp3-minimal layerstree-node-content__options__button layerstree-node-content__options__button--active"
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      title="filter"
      type="button"
    >
      <span
        className="bp3-icon bp3-icon-filter"
        icon="filter"
      >
        <svg
          data-icon="filter"
          height={16}
          viewBox="0 0 16 16"
          width={16}
        >
          <desc>
            filter
          </desc>
          <path
            d="M13.99.99h-12a1.003 1.003 0 0 0-.71 1.71l4.71 4.71V14a1.003 1.003 0 0 0 1.71.71l2-2c.18-.18.29-.43.29-.71V7.41L14.7 2.7a1.003 1.003 0 0 0-.71-1.71z"
            fillRule="evenodd"
          />
        </svg>
      </span>
      <span
        className="bp3-button-text"
      >
        filtres
      </span>
    </button>
    <button
      className="bp3-button bp3-minimal layerstree-node-content__options__button layerstree-node-content__options__button--more"
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      title="options d'affichage"
      type="button"
    >
      <span
        className="bp3-icon bp3-icon-more"
        icon="more"
      >
        <svg
          data-icon="more"
          height={16}
          viewBox="0 0 16 16"
          width={16}
        >
          <desc>
            more
          </desc>
          <path
            d="M2 6.03a2 2 0 1 0 0 4 2 2 0 1 0 0-4zM14 6.03a2 2 0 1 0 0 4 2 2 0 1 0 0-4zM8 6.03a2 2 0 1 0 0 4 2 2 0 1 0 0-4z"
            fillRule="evenodd"
          />
        </svg>
      </span>
      <span
        className="bp3-button-text"
      >
        options
      </span>
    </button>
  </div>,
]
`;
